pkg/apis/core/validation/validation.go

```diff
@@ -425,6 +426,15 @@ func ValidateVolumes(volumes []core.Volume, podMeta *metav1.ObjectMeta, fldPath
 			allErrs = append(allErrs, field.Invalid(idxPath.Child("persistentVolumeClaim").Child("claimName"), vol.PersistentVolumeClaim.ClaimName,
 				"must not reference a PVC that gets created for an ephemeral volume"))
 		}
+		if vol.PersistentVolumeClaim != nil {
+			claimName := vol.PersistentVolumeClaim.ClaimName
+			if _, ok := referPVCs[claimName]; !ok {
+				referPVCs[claimName] = struct{}{}
+			} else {
+				allErrs = append(allErrs, field.Invalid(idxPath.Child("persistentVolumeClaim").Child("claimName"), vol.PersistentVolumeClaim.ClaimName,
```

Comment:
Making this a validation error will turn any existing pods with this configuration from pods that cannot run to pods that cannot be deleted if they have a finalizer (since validation will fail on the update that would try to remove the finalizer).

This has to tolerate updates to pods which already contain this invalid configuration.

The standard way to do this is to add a boolean to PodValidationOptions that indicates whether duplicate PVC volumes should be allowed, set it to false by default in GetValidationOptionsFromPodSpecAndMeta, and set it to true on update IFF the existing pod spec has duplicate PVC volumes already.

Reply:
Done

Reply:
nit: simplify to this and use field.Duplicate:
```go
if vol.PersistentVolumeClaim != nil && !opts.AllowDuplicatePVCNames {
	claimName := vol.PersistentVolumeClaim.ClaimName
	if _, exists := referPVCs[claimName]; exists {
		allErrs = append(allErrs, field.Duplicate(...))
	}
	referPVCs[claimName] = struct{}{}
}
```